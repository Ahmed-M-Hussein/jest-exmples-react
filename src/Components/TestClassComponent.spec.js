import React from "react";
import { create, act } from "react-test-renderer";
import DemoComponent from "../Demos/DemoComponent";

describe("Test Class Component", () => {
  it("renders name Ahmed correctly", () => {
    const tree = create(<DemoComponent name="Ahmed" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders name Guest correctly", () => {
    const tree = create(<DemoComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("test function handlePressBtn inside DemoComponent", () => {
    const wrapper = create(<DemoComponent />);
    const instance = wrapper.getInstance();
    instance.handlePressBtn();
    expect(instance.state.counter).toBe(1);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("test function handlePressBtn 2 times inside DemoComponent", () => {
    const wrapper = create(<DemoComponent />);
    const instance = wrapper.getInstance();

    instance.handlePressBtn();
    instance.handlePressBtn();
    expect(instance.state.counter).toBe(2);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("test  inside DemoComponent", () => {
    let wrapper;
    act(() => {
      wrapper = create(<DemoComponent />);
    });

    expect(wrapper).toMatchSnapshot();

    act(() => {
      wrapper.update(<DemoComponent name="Ahmed" />);
    });

    expect(wrapper).toMatchSnapshot();

    act(() => {
      wrapper.update(<DemoComponent name="Mohamed" />);
    });

    expect(wrapper).toMatchSnapshot();
  });
});
