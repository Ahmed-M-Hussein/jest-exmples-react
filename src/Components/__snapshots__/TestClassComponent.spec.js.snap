// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Class Component renders name Ahmed correctly 1`] = `
<View>
  <Text>
     Welcom 
    Ahmed
     
  </Text>
  <Text>
     You Clicked this Btn 
    0
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test Class Component renders name Guest correctly 1`] = `
<View>
  <Text>
     Welcom 
    Guest
     
  </Text>
  <Text>
     You Clicked this Btn 
    0
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test Class Component test  inside DemoComponent 1`] = `
<View>
  <Text>
     Welcom 
    Guest
     
  </Text>
  <Text>
     You Clicked this Btn 
    0
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test Class Component test  inside DemoComponent 2`] = `
<View>
  <Text>
     Welcom 
    Ahmed
     
  </Text>
  <Text>
     You Clicked this Btn 
    0
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test Class Component test  inside DemoComponent 3`] = `
<View>
  <Text>
     
    Ahmed
     has logged out 
  </Text>
  <Text>
     Welcom 
    Mohamed
     
  </Text>
  <Text>
     You Clicked this Btn 
    0
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test Class Component test function handlePressBtn 2 times inside DemoComponent 1`] = `
<View>
  <Text>
     Welcom 
    Guest
     
  </Text>
  <Text>
     You Clicked this Btn 
    2
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test Class Component test function handlePressBtn inside DemoComponent 1`] = `
<View>
  <Text>
     Welcom 
    Guest
     
  </Text>
  <Text>
     You Clicked this Btn 
    1
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;
